server/src/main/java/org/apache/iotdb/db/utils/QueryUtils.java

```diff
@@ -48,7 +48,7 @@ public static void modifyChunkMetaData(List<ChunkMetadata> chunkMetaData,
     for (int metaIndex = 0; metaIndex < chunkMetaData.size(); metaIndex++) {
       ChunkMetadata metaData = chunkMetaData.get(metaIndex);
       for (Modification modification : modifications) {
-        if (modification.getVersionNum() > metaData.getVersion()) {
+        if (modification.getFileOffset() > metaData.getOffsetOfChunkHeader()) {
```

Comment:
Can modification.getFileOffset() == metaData.getOffsetOfChunkHeader() ?   What should we do in this condition

Reply:
1. metaData.getOffsetOfChunkHeader() marks the byte offset of this chunk header.
2. modification.getFileOffset() returns the offset of any chunk in the file after the chunk has been flushed. 
So I consider it could be guaranteed that if any chunk is successfully flushed, offset in (2) > (1).
Thats why I didn't add the equality condition.

Reply:
Reasonable! So just add some comment here and add test which I said above.

Reply:
You can also check sonar code smell. There is an unused parameter.

Reply:
Thanks, I'll add some more test scenarios.